src/fontinfo.rs

```diff
@@ -0,0 +1,940 @@
+    pub postscript_blue_values: Option<Vec<IntegerOrFloat>>,
+    pub postscript_other_blues: Option<Vec<IntegerOrFloat>>,
+    pub postscript_family_blues: Option<Vec<IntegerOrFloat>>,
+    pub postscript_family_other_blues: Option<Vec<IntegerOrFloat>>,
+    pub postscript_stem_snap_h: Option<Vec<IntegerOrFloat>>,
+    pub postscript_stem_snap_v: Option<Vec<IntegerOrFloat>>,
+    pub postscript_blue_fuzz: Option<IntegerOrFloat>,
+    pub postscript_blue_shift: Option<IntegerOrFloat>,
+    pub postscript_blue_scale: Option<Float>,
+    pub postscript_force_bold: Option<bool>,
+    pub postscript_default_width_x: Option<IntegerOrFloat>,
+    pub postscript_nominal_width_x: Option<IntegerOrFloat>,
+    pub postscript_weight_name: Option<String>,
+    pub postscript_default_character: Option<String>,
+    pub postscript_windows_character_set: Option<PostscriptWindowsCharacterSet>,
+
+    // Macintosh FOND Resource Data
+    #[serde(rename = "macintoshFONDFamilyID")]
+    pub macintosh_fond_family_id: Option<Integer>,
+    #[serde(rename = "macintoshFONDName")]
+    pub macintosh_fond_name: Option<String>,
+
+    // WOFF Data
+    pub woff_major_version: Option<NonNegativeInteger>,
+    pub woff_minor_version: Option<NonNegativeInteger>,
+    #[serde(rename = "woffMetadataUniqueID")]
+    pub woff_metadata_unique_id: Option<WoffMetadataUniqueID>,
+    pub woff_metadata_vendor: Option<WoffMetadataVendor>,
+    pub woff_metadata_credits: Option<WoffMetadataCredits>,
+    pub woff_metadata_description: Option<WoffMetadataDescription>,
+    pub woff_metadata_license: Option<WoffMetadataLicense>,
+    pub woff_metadata_copyright: Option<WoffMetadataCopyright>,
+    pub woff_metadata_trademark: Option<WoffMetadataTrademark>,
+    pub woff_metadata_licensee: Option<WoffMetadataLicensee>,
+    pub woff_metadata_extensions: Option<Vec<WoffMetadataExtensionRecord>>,
+}
+
+impl FontInfo {
+    /// Validates various fields according to the [specification][].
+    ///
+    /// [specification]: http://unifiedfontobject.org/versions/ufo3/fontinfo.plist/
+    pub fn validate(&self) -> Result<(), Error> {
+        // unitsPerEm must be non-negative.
+        if let Some(v) = self.units_per_em {
+            if v.is_sign_negative() {
+                return Err(Error::FontInfoError);
+            }
+        }
+
+        // The date format is "YYYY/MM/DD HH:MM:SS". This does not validate that the
+        // days ceiling is valid for the month, as this would probably need a specialist
+        // datetime library.
+        if let Some(v) = &self.open_type_head_created {
+            const DATE_LENGTH: usize = 19;
+            if v.len() != DATE_LENGTH {
+                return Err(Error::FontInfoError);
+            }
+            if !v.chars().all(|b| b.is_ascii_digit() || b == ' ' || b == '/' || b == ':') {
+                return Err(Error::FontInfoError);
+            }
+
+            if !(v[0..4].parse::<u16>().is_ok()
```

Comment:
for all these parse errors we _could_ do something like, `x.parse::<u8>().map_err(|_| Error::FontInfoError)?`